esp_at_lib/src/include/esp/esp_includes.h

```diff
@@ -35,7 +35,8 @@
 #define ESP_HDR_INCLUDES_H
 
 #ifdef __cplusplus
-extern "C" {
+// don't nest extern "C"
+// extern "C" {
```

Comment:
Nesting of extern is OK, and just make sure it is always linked as C

Reply:
Sorry for this late reply.
I know you method could worked well most cases.
A error condition for Example.
extern "C" {
#include "third_party.h"
}
will report error, if third_party.h is a C++ header, 
include class keywords which C can't recognized.

If we always let #include outside extern "C", and let every header owner make he's declarition as C linkage, we could avoid above problems.

Refer: http://effective-c.googlecode.com/svn-history/r67/wiki/EffectiveC.wiki

Reply:
turnary, you should NOT modify esp files.

The problem might be that extern C shall be after all includes. This I will review.

PS: Your link doesn't work.